lib/game_logic.rb

```diff
@@ -0,0 +1,78 @@
+    puts '-------------'
+    puts "| #{@array[6]} | #{@array[7]} | #{@array[8]} |"
+    puts '-------------'
+  end
+
+  def validate_move(move)
+    @arrays_key = { 'a' => 0, 'b' => 1, 'c' => 2, 'd' => 3, 'e' => 4, 'f' => 5, 'g' => 6, 'h' => 7, 'i' => 8 }
+    @arrays_key.key?(move) ? @arrays_key[move] : 'No Valid'
+  end
+
+  def update_board(move, symbol)
+    if @array[move] != 'X' && @array[move] != 'O'
+      @array[move] = symbol
+      true
+    else
+      puts 'Move is Invalid'
+      false
+    end
+  end
+
+  def winning?
+    return true if @array[0] == @array[1] && @array[1] == @array[2]
+    return true if @array[3] == @array[4] && @array[4] == @array[5]
+    return true if @array[6] == @array[7] && @array[7] == @array[8]
+    return true if @array[0] == @array[3] && @array[3] == @array[6]
+    return true if @array[1] == @array[4] && @array[4] == @array[7]
+    return true if @array[2] == @array[5] && @array[5] == @array[8]
+    return true if @array[0] == @array[4] && @array[4] == @array[8]
+    return true if @array[2] == @array[4] && @array[4] == @array[6]
+    return 'draw' if @array.none? { |position| %w[a b c d e f g h i].any?(position) }
+
+    false
+  end
+
+  def winner(player1, player2)
+    return player1.name if @array[0] == @array[1] && @array[1] == @array[2] && array[0] == player1.symbol
+    return player1.name if @array[6] == @array[7] && @array[7] == @array[8] && array[6] == player1.symbol
+    return player1.name if @array[3] == @array[4] && @array[4] == @array[5] && array[3] == player1.symbol
+    return player1.name if @array[0] == @array[3] && @array[3] == @array[6] && array[0] == player1.symbol
+    return player1.name if @array[1] == @array[4] && @array[4] == @array[7] && array[1] == player1.symbol
+    return player1.name if @array[2] == @array[5] && @array[5] == @array[8] && array[2] == player1.symbol
+    return player1.name if @array[0] == @array[4] && @array[4] == @array[8] && array[0] == player1.symbol
+    return player1.name if @array[2] == @array[4] && @array[4] == @array[6] && array[2] == player1.symbol
+
+    return player2.name if @array[6] == @array[7] && @array[7] == @array[8] && array[6] == player2.symbol
+    return player2.name if @array[0] == @array[1] && @array[1] == @array[2] && array[0] == player2.symbol
+    return player2.name if @array[3] == @array[4] && @array[4] == @array[5] && array[3] == player2.symbol
+    return player2.name if @array[0] == @array[3] && @array[3] == @array[6] && array[0] == player2.symbol
+    return player2.name if @array[1] == @array[4] && @array[4] == @array[7] && array[1] == player2.symbol
+    return player2.name if @array[2] == @array[5] && @array[5] == @array[8] && array[2] == player2.symbol
+    return player2.name if @array[0] == @array[4] && @array[4] == @array[8] && array[0] == player2.symbol
+    return player2.name if @array[2] == @array[4] && @array[4] == @array[6] && array[2] == player2.symbol
```

Comment:
- [ ] This is also not respecting the DRY rule and not efficient. In fact, there will not be any need for this logic, if you successfully implement the requirement I stated in here.

![image](https://user-images.githubusercontent.com/50773261/88352547-0d28a000-cd52-11ea-8e84-2788fe21362b.png)

Kindly remove them after implementing the above requirement.